src/main/java/core/basesyntax/model/lib/Injector.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.model.lib;
+
+import core.basesyntax.model.dao.BetDao;
+import core.basesyntax.model.dao.UserDao;
+import core.basesyntax.model.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoDaoAnnotationException {
+
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            field.setAccessible(true);
+            if (field.getAnnotation(Inject.class) != null) {
+                if (checkBetDaoType(field)) {
+                    field.set(instance, Factory.getBetDao());
+                }
+                if (checkUserDaoType(field)) {
+                    field.set(instance, Factory.getUserDao());
+                }
+            }
+        }
+        return instance;
+    }
+
+    private static boolean checkBetDaoType(Field field) {
```

Comment:
let's maybe try to write this elegantly without static helper methods

Reply:
done

Reply:
done